server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -583,7 +583,7 @@ public void handshake(
                 } else {
                     l.onResponse(response);
                 }
-            }), HandshakeResponse::new, ThreadPool.Names.GENERIC)
+            }), HandshakeResponse::new, ThreadPool.Names.CLUSTER_COORDINATION)
```

Comment:
I see your point, but also I don't like adding that parameter here when it'll always be `ThreadPool.Names.CLUSTER_COORDINATION`, that's weird also. Isn't the probably that the handshake stuff just doesn't belong in the `TransportService`?
Seems like that whole codebase could just live somewhere outside of the transport service since it's really more of cluster coordination thing. WDYT about that kind of refactoring in a follow up?

Reply:
🤦 right ... lets do the parameter thing then :)